pg_search/src/query/heap_field_filter.rs

```diff
@@ -168,9 +121,9 @@ impl HeapFieldFilter {
         }
 
         // Use provided expression context if available, otherwise create a standalone one
-        let (econtext, should_free_context) = if !provided_expr_context.is_null() {
+        let (econtext, standalone_context) = if let Some(ctx) = expr_context {
             // Use the provided context (supports subqueries)
-            (provided_expr_context, false)
+            (ctx.as_ptr(), None)
         } else {
             // Create a standalone context (fallback for simple expressions)
             let standalone_context = pg_sys::CreateStandaloneExprContext();
```

Comment:
Can't comment on the line below, but: it feels like we would want to panic if this were null, rather than returning `false`? If not, a comment explaining why would be good.

Reply:
Panic is a good idea. Done.